app/models/manageiq/providers/vmware/infra_manager/vm/reconfigure.rb

```diff
@@ -239,6 +253,32 @@ def add_disk_config_spec(vmcs, options)
     end
   end
 
+  def add_network_adapter_config_spec(vmcs, options)
+    add_device_config_spec(vmcs, VirtualDeviceConfigSpecOperation::Add) do |vdcs|
+      vdcs.device = VimHash.new("VirtualVmxnet3") do |dev|
+        dev.key = rand(-9999..-100) # negative integer as temporary key
+        dev.unitNumber = 0
```

Comment:
I _think_ you need to pick a unitNumber that is available and give it a controllerKey as well.  For the key we usually just use `-100 * options[:unit_number]`

Reply:
Using unitNumber 0 instructs VMware to choose the next available unit number. The key must be a unique negative number (temporary key). The controllerKey will be chosen by VMware.